### PR TITLE
Replace boost::multi_index_container with tbb::concurrent_unordered_set in symtab_impl

### PR DESCRIPTION
**Pull request by hainest**
There are now only two dimensions to each Module, so the multi_index isn't needed. This also replaces the mutex with the intrinsic reader/writer locks in TBB.

### CODE REVIEW

File: symtabAPI/src/indexed_modules.hpp
Comment (kupsch) [inline symtabAPI/src/indexed_modules.hpp:1]: add copyright and header guard

File: symtabAPI/src/Symtab.C [hunk 1]
```diff
@@ -504,10 +504,10 @@ bool Symtab::fixSymModules(std::vector<Symbol *> &raw_syms)
 	createDefaultModule();
     }
 
-    for (auto i = impl->indexed_modules.begin(); i != impl->indexed_modules.end(); ++i)
+    for (auto *m : this->impl->modules)
```
Comment (kupsch) [inline symtabAPI/src/Symtab.C:507]: remove the `this->` for consistentcy

**Review by kupsch — APPROVED** [review]
in another PR (or here) change the `.hpp` found in the PR and the other two to `.h` files to be consistent with the rest of dyninst.  Couple of other minor comments below

File: symtabAPI/src/indexed_modules.hpp
Comment (hainest) [reply symtabAPI/src/indexed_modules.hpp:1]: I always forget these...

File: symtabAPI/src/Symtab.C (hunk 1 above)
Comment (hainest) [reply symtabAPI/src/Symtab.C]: No idea how that got there. Fixed.

**Comment by hainest** [comment]
> in another PR (or here) change the `.hpp` found in the PR and the other two to `.h` files to be consistent with the rest of dyninst. Couple of other minor comments below

Fixed indexed_modules here. I'll submit a separate PR for the other two.

**Pull request merged**